hazelcast/src/main/java/com/hazelcast/cardinality/impl/hyperloglog/impl/SparseHyperLogLogEncoder.java

```diff
@@ -103,7 +108,10 @@ public int getId() {
     public void writeData(ObjectDataOutput out) throws IOException {
         mergeAndResetTmp();
         out.writeInt(p);
-        out.writeInt(pPrime);
+        // RU_COMPAT_3_9
+        if (out.getVersion().isLessThan(V3_10)) {
+            out.writeInt(Integer.MIN_VALUE);
```

Comment:
AFAIU, this should not affect the consumers of the written data since p' is always 25, and it was different only in tests, right?

Reply:
Yes it won't affect anyone, it helps debugging in case sth is wrong and we get a heap dump, we will be able to see the MIN_VALUE in the stream rather than the 25. Other than that, there is no other `precision` value used anywhere, neither in tests; I mean for the SPARSE.

Reply:
Is it possible to have a situation when the cardinality estimator is migrated back to the 3.9 member during RO? Seems like in this case `SparseHyperLogLogEncoder#init` would receive a totally unexpected p' value of -1.

Reply:
Yes, you are right, I thought I had it  `greaterThanOrEqual`. Just fixed it, good catch.